src/Acquisition.jl

```diff
@@ -35,4 +37,43 @@ include("plot.jl")
 include("calc_powers.jl")
 include("est_signal_noise_power.jl")
 include("acquire.jl")
+
+
+@setup_workload begin
+    # Putting some things in `@setup_workload` instead of `@compile_workload` can reduce the size of the
+    # precompile file and potentially make loading faster.
+    rate = 2.048e6 
+
+    #common real data
+    data_i8 = rand(Int8,Int(round(rate*0.01)))
+    data_i16 = rand(Int16,Int(round(rate*0.01)))
+    data_f32 = rand(Float32,Int(round(rate*0.01)))
+    data_f64 = rand(Float64,Int(round(rate*0.01)))
+
+
+    #common complex data
+    data_ci8 = rand(Complex{Int8},Int(round(rate*0.01)))
+    data_ci16 = rand(Complex{Int16},Int(round(rate*0.01)))
+    data_cf32 = rand(Complex{Float32},Int(round(rate*0.01)))
+    data_cf64 = rand(Complex{Float64},Int(round(rate*0.01)))
+
+
+    @compile_workload begin
+        # all calls in this block will be precompiled, regardless of whether
+        # they belong to your package or not (on Julia 1.8 and higher)
+        #TODO add support for other GNSS constellations
+        samplerate = rate*Hz
+        acquire(GPSL1(),data_ci8[1:2048], samplerate, 1:32; interm_freq=0*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_ci16[1:2048], samplerate, 1:32; interm_freq=0*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_cf32[1:2048], samplerate, 1:32; interm_freq=0*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_cf64[1:2048], samplerate, 1:32; interm_freq=0*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_i8[1:2048], samplerate, 1:32; interm_freq=10*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_i16[1:2048], samplerate, 1:32; interm_freq=10*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_f32[1:2048], samplerate, 1:32; interm_freq=10*Hz, max_doppler=10e3*Hz);
+        acquire(GPSL1(),data_f64[1:2048], samplerate, 1:32; interm_freq=10*Hz, max_doppler=10e3*Hz);
```

Comment:
```suggestion
        foreach(systems) do system
            foreach(signals) do signal
                foreach(samplerates) do samplerate
                    foreach(interm_freqs) do interm_freq
                        foreach(max_dopplers) do max_doppler
                            acquire(system, signal, samplerate, 1:1; interm_freq, max_doppler)
                        end
                    end
                end
            end
        end
```
(It is enough to acquire a single satellite)

I haven't tested the code, though ;)
You can delete the comments, IMHO